File: TypeScript/day01/part2.ts
import { getPuzzleInput } from "../utils";
const input = getPuzzleInput("day01");

export default function part2(input: string) {
  return input
    .split("\n\n")
    .map((elf) =>
      elf
        .split("\n")
        .map((item) => Number(item))
        .reduce((total, current) => total + current)
    )
    .sort((a, b) => b - a)
    .slice(0, 3)
    .reduce((total, current) => total + current);
}

File: Go/day01/part2.go
package main

import (
	"adventofcode/packages/utils"
	"sort"
	"strconv"
)

func part2() int {
	contentSlice := utils.GetPuzzleInput("day01")
	groups := make([][]int, 0)
	current := make([]int, 0)
	for _, v := range contentSlice {
		num, err := strconv.Atoi(v)
		if err != nil {
			groups = append(groups, current)
			current = make([]int, 0)
		} else {
			current = append(current, num)
		}
	}

	t := make([]int, 0)
	for _, v := range groups {
		t = append(t, utils.Sum(v))
	}

	sort.Sort(sort.Reverse(sort.IntSlice(t)))
	t3 := t[0:3]
	return utils.Sum(t3)
}
